### ISSUE TITLE
[Wayland] error: 'm_XWindow' was not declared in this scope

### ISSUE THREAD

**Issue by SupervisedThinking** [moonlight-stream/moonlight-qt#680]
Well I try to build the latest release with wayland + vaapi -> `HAVE_LIBVA_WAYLAND` is defined, SDL2 & VAAPI were build with Wayland support.

I guess something went wrong here: 
https://github.com/moonlight-stream/moonlight-qt/blob/master/app/streaming/video/ffmpeg-renderers/vaapi.cpp#L61-L62

`m_XWindow` is declared here:
https://github.com/moonlight-stream/moonlight-qt/blob/master/app/streaming/video/ffmpeg-renderers/vaapi.h#L60-L62

**Describe the bug**
```
In file included from /mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.h:3,
                 from /mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp:3:
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/renderer.h:14: warning: "EGL_NO_X11" redefined
   14 | #define EGL_NO_X11
      | 
<command-line>: note: this is the location of the previous definition
In file included from /mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp:3:
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.h:7:2: warning: #warning Unable to use libva-x11 without SDL support [-Wcpp]
    7 | #warning Unable to use libva-x11 without SDL support
      |  ^~~~~~~
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp: In member function 'void* VAAPIRenderer::openDisplay(SDL_Window*)':
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp:62:9: error: 'm_XWindow' was not declared in this scope
   62 |         m_XWindow = info.info.x11.window;
      |         ^~~~~~~~~
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp:62:31: error: 'union SDL_SysWMinfo::<unnamed>' has no member named 'x11'
   62 |         m_XWindow = info.info.x11.window;
      |                               ^~~
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp: In member function 'virtual void VAAPIRenderer::renderFrame(AVFrame*)':
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp:353:17: warning: unused variable 'surface' [-Wunused-variable]
  353 |     VASurfaceID surface = (VASurfaceID)(uintptr_t)frame->data[3];
      |                 ^~~~~~~
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/streaming/video/ffmpeg-renderers/vaapi.cpp:355:27: warning: unused variable 'vaDeviceContext' [-Wunused-variable]
  355 |     AVVAAPIDeviceContext* vaDeviceContext = (AVVAAPIDeviceContext*)deviceContext->hwctx;
      |                           ^~~~~~~~~~~~~~~
make[2]: *** [Makefile.Release:1272: release/vaapi.o] Error 1
make[2]: Leaving directory '/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/.x86_64-libreelec-linux-gnu/app'
make[1]: *** [Makefile:42: release] Error 2
make[1]: Leaving directory '/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/.x86_64-libreelec-linux-gnu/app'
make: *** [Makefile:102: sub-app-make_first] Error 2
FAILURE: scripts/build moonlight-qt during make_target (default)
*********** FAILED COMMAND ***********
make ${PKG_MAKE_OPTS_TARGET}
**************************************
```

**Steps to reproduce**
Build config:
```
/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/toolchain/bin/qmake -o Makefile /mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/build/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/app/app.pro 'CONFIG+=embedded HAVE_LIBVA_WAYLAND=1' PREFIX=/mnt/dev/LibreELEC-RR/build.LibreELEC-wayland.x86_64-11.0-devel/install_pkg/moonlight-qt-6c523570e5cfda8b309f193bb3d12682df8645df/usr ) && make -f Makefile 
Project MESSAGE: FFmpeg decoder selected
Project MESSAGE: VAAPI renderer selected
Project MESSAGE: VAAPI Wayland support enabled
Project MESSAGE: VAAPI DRM support enabled
Project MESSAGE: DRM renderer selected
Project MESSAGE: Master hooks enabled
Project MESSAGE: EGL renderer selected
Project MESSAGE: Embedded build
Project MESSAGE: FFmpeg decoder selected
Project MESSAGE: VAAPI renderer selected
Project MESSAGE: VAAPI Wayland support enabled
Project MESSAGE: VAAPI DRM support enabled
Project MESSAGE: DRM renderer selected
Project MESSAGE: Master hooks enabled
Project MESSAGE: EGL renderer selected
Project MESSAGE: Embedded build
Project MESSAGE: FFmpeg decoder selected
Project MESSAGE: VAAPI renderer selected
Project MESSAGE: VAAPI Wayland support enabled
Project MESSAGE: VAAPI DRM support enabled
Project MESSAGE: DRM renderer selected
Project MESSAGE: Master hooks enabled
Project MESSAGE: EGL renderer selected
Project MESSAGE: Embedded build
```

**Comment by cgutman** [comment]
This should be fixed by 0fa7723f510d1b09f11362126e3f9a708658ee3d.

Just to be sure, you are intending to create a Moonlight build that can only run under Wayland? The build failure was due to missing X11 support in libva and SDL, which is a pretty uncommon situation.

**Comment by SupervisedThinking** [comment]
Yes that's the exact intention -> I build embedded images that either target GBM, Wayland or X11 for x86_64 😃 Each image only ships Kodi (GBM) or Kodi build for Wayland or X11. So in this specific case Wayland, libva & SDL were present but no X11 headers.
